### PR TITLE
adding play button on video header articles

### PR DESCRIPTION
**Pull request by leonoragilmour**
image

### CODE REVIEW

File: app/Services/Generator.php [hunk 1]
```diff
@@ -37,7 +37,10 @@ public function create_object( \WP_Post $post ) {
 		$object->metadata->dateModified = gmdate('c', strtotime($post->post_modified));
 		$object->metadata->authors = [ $author->display_name ];
 		$object->metadata->excerpt = get_post_meta( $post->ID, 'sell', true );
-		$object->metadata->thumbnailURL = $this->get_representative_article_image_url( $post );
+
+		$video = $this->get_video_url( $post );
+		$object->metadata->videoURL = $video['video_url'];
```
Comment (jonsherrard) [inline app/Services/Generator.php]: Should we conditionally set these values? Or do they always need to be there?

I.e `if ($video) { // do stuff }`

File: app/Services/Generator.php [hunk 2]
```diff
@@ -119,6 +122,16 @@ private function get_representative_article_image_url( $post ) {
 		)[0];
 	}
 
+	private function get_video_url( $post ) {
+		if(get_post_meta( $post->ID, 'header_type', true ) !== 'hero-with-video') {
+			return null;
+		}
+		return [
+			"video_url" => "https://cdn.jwplayer.com/manifests/" . get_post_meta( $post->ID, 'header_video_portrait_id', true ) . ".m3u8",
+			"still_url" => "https://cdn.jwplayer.com/thumbs/" . get_post_meta( $post->ID, 'header_video_portrait_id', true ) . "-720.jpg",
```
Comment (jonsherrard) [inline app/Services/Generator.php]: Wicked! 👍 👍

File: app/Services/Generator.php (hunk 1 above)
Comment (leonoragilmour) [reply app/Services/Generator.php]: Well I would've done that but Romulo doesn't like `else`s and I would've needed one for the thumbnail 😅

File: app/Services/Generator.php (hunk 2 above)
Comment (leonoragilmour) [reply app/Services/Generator.php]: this is stupid. I'm callling twice... hang on

File: app/Services/Generator.php (hunk 1 above)
Comment (jonsherrard) [reply app/Services/Generator.php]: `$video ? $video['video_url']; : null;

To avoid the notice being thrown?

**Pull request merged**